arctic/store/version_store.py

```diff
@@ -576,6 +575,135 @@ def write(self, symbol, data, metadata=None, prune_previous_version=True, **kwar
         return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=version['version'],
                              metadata=version.pop('metadata', None), data=None)
 
+    def _add_new_version_using_reference(self, symbol, new_version, reference_version, prune_previous_version):
+        if not new_version or \
+                not reference_version or \
+                new_version['symbol'] != reference_version['symbol'] or \
+                new_version['_id'] == reference_version['_id'] or \
+                not new_version['base_version_id']:
+            raise ArcticException("Illegal or non-matching nwe_version or reference_version were supplied: %s and %s",
+                                  new_version, reference_version)
+
+        # Check if in the meanwhile the reference version (based on which we updated incrementally) has been removed
+        last_look = self._versions.find_one({'_id': reference_version['_id']})
+        if last_look is None or last_look.get('deleted'):
+            raise pymongo.errors.OperationFailure("Failed to write metadata for symbol %s. "
+                                                  "The previous version (%s, %d) has been removed during the update" %
+                                                  (symbol, str(reference_version['_id']), reference_version['version']))
+
+        # There is always a small risk here another process in between these two calls (above/below)
+        # to delete the reference_version, which may happen to be the last parent entry in the data segments.
+        # In this case the segments will be deleted by the other process,
+        # and the new version's "base_version_id" won't be referenced by any segments.
+
+        # Insert the new version into the version DB
+        # (must come before the pruning, otherwise base version won't be preserved)
+        mongo_retry(self._versions.insert_one)(new_version)
+
+        if prune_previous_version and reference_version:
+            self._prune_previous_versions(symbol)
+
+        logger.debug('Finished updating versions with new metadata for %s', symbol)
+
+        self._publish_change(symbol, new_version)
+
+        return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=new_version['version'],
+                             metadata=new_version.pop('metadata', None), data=None)
+
+    def write_metadata(self, symbol, metadata, prune_previous_version=True, **kwargs):
```

Comment:
I still think `update_metadata(...)` would be clearer, but happy to go with the consensus on this.

Reply:
(and fwiw I think `write_` is more consistent with this and the other APIs...)

Reply:
I'd also vote for write_* simply for consistency, while I like more the update semantically.